test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1726,6 +1728,40 @@ UCS_TEST_P(test_ucp_sockaddr_protocols, am_zcopy_64k,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols)
 
 
+class test_ucp_sockaddr_protocols_diff_config : public test_ucp_sockaddr_protocols
+{
+public:
+    void init() {
+        m_err_count = 0;
+        get_sockaddr();
+        test_base::init();
+        // the entities will be created in the test
+    }
+};
+
+
+UCS_TEST_P(test_ucp_sockaddr_protocols_diff_config, diff_num_paths)
+{
+    m_env.push_back(new ucs::scoped_setenv("UCX_IB_NUM_PATHS", "1"));
```

Comment:
set also UCX_MAX_EAGER_LANES=2

Reply:
you mean only for when UCX_IB_NUM_PATHS=1 ?

Reply:
like this
https://github.com/openucx/ucx/blob/728dda3bd6600f838b026dc76878d4c66ac3a1cd/test/gtest/ucp/test_ucp_wireup.cc#L1434
pushing to m_envs makes sens only in init() function / ctor when it's outside single function's scope

Reply:
like this. no need to push to m_env
```
UCS_TEST_P() {
scoped_setenv(UCX_MAX_EAGER_LANES=2)
{
  scoped_setenv(PATHS=1)
  create_ent
}
{
  scoped_setenv(PATHS=2)
  create_ent
}
...
}

Reply:
in order to test 2 msg sizes, i moved the entities creation to the init function and the test would only call test_tag_send_recv with a certain msg size. then can keep it as it was?

Reply:
still need to have 2 different scopes like described above because we set the same env var to different values